src/Mono.Android/Android.Runtime/InputStreamInvoker.cs

```diff
@@ -92,7 +98,22 @@ public override int Read (byte[] buffer, int offset, int count)
 
 		public override long Seek (long offset, SeekOrigin origin)
 		{
-			throw new NotSupportedException ();
+			if (BaseFileChannel is object) {
+				switch (origin) {
+				case SeekOrigin.Begin:
+					BaseFileChannel.Position (offset);
+					break;
+				case SeekOrigin.Current:
+					BaseFileChannel.Position (BaseFileChannel.Position() + offset);
+					break;
+				case SeekOrigin.End:
+					BaseFileChannel.Position (BaseFileChannel.Size() + offset);
+					break;
+				}
+				return BaseFileChannel.Position ();
+			} else {
+				throw new NotSupportedException ();
```

Comment:
Style convention: short blocks should go first, as that can take advantage of early exits and reduce indentation levels:

```csharp
if (BaseFileChannel == null)
    throw new NotSupportedException ();
switch (origin) {
    ...
}
```